bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java

```diff
@@ -159,12 +161,29 @@ private ProcessInstance getProcessFromRequest(SlingHttpServletRequest request) {
 
     List<String> ignoredInputs = Arrays.asList("definition", "description", "action");
 
-    private Map<String, Object> convertRequestMap(Map<String, String[]> parameterMap) {
-        return parameterMap.entrySet().stream()
+    private Map<String, Object> convertRequestMap(RequestParameterMap requestParameterMap) {
+        return requestParameterMap.entrySet().stream()
                 .filter(entry -> !ignoredInputs.contains(entry.getKey()))
                 .collect(Collectors.toMap(
                         entry -> entry.getKey(),
-                        entry -> (Object) (entry.getValue().length == 1 ? entry.getValue()[0] : entry.getValue())
+                        entry -> {
+                            final RequestParameter[] values = entry.getValue();
```

Comment:
@badvision Changed this to handle file uploads.

Reply:
Oh, good thinking!  Just keep in mind it was designed such that single value = object, array => Array of objects, etc.  I haven't hooked up a component that generates multiple values as inputs into the servlet so this was an area of future expansion for multivalue fields, etc.

Reply:
Not sure I follow this ...